docs/WindowsBuild.md

```diff
@@ -106,303 +87,715 @@ Set up the `ucrt`, `visualc`, and `WinSDK` modules by:
+  -G Ninja ^
+  -S S:\llvm-project\llvm
+
+cmake --build S:\b\1
+cmake --build S:\b\1 --target install
+```
+
+## Build for development
+
+The following guide will get you through the building process of a complete Swift debug toolchain.
+
+Before you kick off the process, unset `%SDKROOT%` if you've already installed Swift:
+
+```cmd
+set SDKROOT=
+```
+
+If you want to use an existing Swift toolchain to build the core libraries and other parts of the toolchain, make sure you have it in `%Path%`, and strip `S:/b/1/bin/` from every `-D CMAKE_{language}_COMPILER=S:/b/1/bin/{compiler}` option.
+
+Here is a graphical overview of all build steps from this guide.
+
+```
+       ┌──────────────────────────────────────┐      ┌────────────────┐
+       │                                      │      │                │
+       │       1. LLVM + Swift compiler       │      │                │
+       │                                      │      │ Existing Swift │
+       │    ┌────────────────────────────┐    │  or  │   toolchain    │
+       │    │ 2. Swift runtime libraries │    │      │                │
+       │    └────────────────────────────┘    │      │                │
+       └──────────────────────────────────────┘      └────────────────┘
+                           │                                  │
+                           └─────────┬────────────────────────┘
+                                     │
+                 ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─
+                               ┌──────────────────────┐ │
+                 │             │    3. libdispatch    │
+                               └──────────────────────┘ │
+                 │     Core    ┌──────────────────────┐
+               ┌─              │    4. Foundation     │ │─┐
+               │ │  libraries  └──────────────────────┘   │
+               │               ┌──────────────────────┐ │ │
+               │ │             │      5. XCTest       │   │
+               │               └──────────────────────┘ │ │
+               │ └ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─  │
+               │                                          │
+               │                      ┌──────────────────────────────────────┐
+               │                      │                                      │
+               │                      │      7. Toolchain dependencies       │
+┌────────────────────────────┐        │                                      │
+│         6. LLBuild         │        ├────┬───────────┬────────┬────────────┤
+└────────────────────────────┘        │Yams│SwiftSystem│  ···   │IndexStoreDB│
+               │                      └────┴───────────┴────────┴────────────┘
+               │                                          │
+               │                                          │
+               │                           ┌────────────────────────────┐
+               │                           │           8. TSC           │
```

Comment:
I moved TSC out because these dependencies have no dependency on each other, so they can be built in any order, and optionally (eg. you can skip IndexStoreDB if you’re not building SourceKit-LSP).

However, this is not the case for TSC now. TSC depends on SwiftSystem by the time, and it’s also depended by every developer tool we ship. So we definitely need to distinguish it from other dependencies.